udp-reassembly-offload.md

```diff
@@ -38,11 +39,13 @@ URO can only be attempted on a batch of packets that meet **all** the following
 
 - 5-tuple matches.
 - Payload length is identical for all datagrams, except the last datagram which may be less.
-- The checksums on pre-coalesced packets must be correct.
-- ECN, DF bits must match on all packets (IPv4).
-- NextHeader must be UDP (IPv6).
+- The UDP checksums on pre-coalesced packets must be correct.
+- TTL, ToS/ECN, Protocol, and DF bit must match on all packets (IPv4).
+- TC/ECN, FlowLabel, and HopLimit must match, and NextHeader must be UDP (IPv6).
 - The total length of the Single Coalesced Unit (SCU) must not exceed IP max length.
 
+The coalesced IP length field and the UDP length field must be the new coalesced length. The coalesced IPv4 checksum field must include the new length. The coalesced UDP checksum field is ignored and does not need to be calculated (since it was already validated individually).
```

Comment:
One suggestion:

```suggestion
The coalesced IP length field and the UDP length field must reflect the new coalesced length. The coalesced IPv4 checksum field must include the new length. The coalesced UDP checksum field is ignored and does not need to be calculated (since it was already validated individually).
```

> IPv4 checksum field must include the new length

Why do we require them to recalculate the checksum?

Reply:
the IP checksum is just over the IP header, IIRC, and the stack may check that value. (The Software URO code recalculates the IP checksum)

Reply:
Ok. Then let's leave it as you have it for now.